### PR TITLE
docs(modal): add canDismiss docs

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Part of https://github.com/ionic-team/ionic-framework/issues/22297

## What is the new behavior?

- This adds the documentation + usage examples for canDismiss

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/modal/readme.md [hunk 1]
```diff
@@ -221,19 +243,55 @@ When the backdrop is disabled, users will be able to interact with elements outs
     <app-angular-component title="Ionic"></app-angular-component>
   </ng-template>
 </ion-modal>
+
+<!-- Require Action Sheet confirmation before dismissing -->
+<ion-modal [isOpen]="true" [canDismiss]="canDismiss()">
```
Comment (sean-perkins) [inline core/src/components/modal/readme.md:248]:
```suggestion
<ion-modal [isOpen]="true" [canDismiss]="canDismiss">
```

File: core/src/components/modal/readme.md [hunk 2]
```diff
@@ -655,6 +780,11 @@ export const ModalExample: React.FC<Props> = ({ router }) => {
       <IonModal isOpen={true}>
         <AppReactComponent title="Ionic"></AppReactComponent>
       </IonModal>
+      
+      {/* Require Action Sheet confirmation before dismissing */}
+      <IonModal isOpen={true} canDismiss={() => canDismiss()}>
```
Comment (sean-perkins) [inline core/src/components/modal/readme.md:785]: Can this be shorthanded to:
```tsx
canDismiss={canDismiss}
```

File: core/src/components/modal/readme.md [hunk 3]
```diff
@@ -917,6 +1074,11 @@ export class ModalExample {
         <ion-modal isOpen={true}>
           <app-stencil-component title="Ionic"></app-stencil-component>
         </ion-modal>
+        
+        {/* Require Action Sheet confirmation before dismissing */}
```
Comment (sean-perkins) [inline core/src/components/modal/readme.md:1078]: Similar question regarding shorthand syntax.

File: core/src/components/modal/usage/angular.md [hunk 4]
```diff
@@ -44,19 +44,55 @@
     <app-angular-component title="Ionic"></app-angular-component>
   </ng-template>
 </ion-modal>
+
+<!-- Require Action Sheet confirmation before dismissing -->
+<ion-modal [isOpen]="true" [canDismiss]="canDismiss()">
```
Comment (sean-perkins) [inline core/src/components/modal/usage/angular.md:49]:
```suggestion
<ion-modal [isOpen]="true" [canDismiss]="canDismiss">
```

File: core/src/components/modal/readme.md (hunk 3 above)
Comment (liamdebeasi) [reply core/src/components/modal/readme.md:1078]: What's the question?

Comment (sean-perkins) [reply core/src/components/modal/readme.md:1078]: I didn't click on the correct line when reviewing, clearly 😆  on line 1079 I believe, using the shorthand syntax for property binding:
```
canDismiss={canDismiss}
```

Comment (liamdebeasi) [reply core/src/components/modal/readme.md:1078]: Oh okay yeah -- I've updated that 👍

**Review by sean-perkins — APPROVED** [review]

**Review by amandaesmith3 — APPROVED** [review]

**Pull request merged**